### PR TITLE
 clipmenu: automatic selection of CM_LAUNCHER

### PR DESCRIPTION
**Pull request by moviuro**

### CODE REVIEW

File: clipmenu [hunk 1]
```diff
@@ -1,7 +1,17 @@
 #!/bin/bash
 
-: "${CM_LAUNCHER=dmenu}"
 : "${CM_DIR="${XDG_RUNTIME_DIR-"${TMPDIR-/tmp}"}"}"
+# Pick dmenu or rofi, depending on what's available on the system
+if [[ -z "$CM_LAUNCHER" ]]; then
```
Comment (cdown) [inline clipmenu]: if ! [[ -v CM_LAUNCHER ]]; then

File: clipmenu [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
+  if command -v dmenu >/dev/null 2>&1; then
+    CM_LAUNCHER=dmenu
```
Comment (cdown) [inline clipmenu]: indentation in this patch is wrong, it should be 4 spaces per indent.

File: clipmenu [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
+  elif command -v rofi >/dev/null 2>&1; then
+    CM_LAUNCHER=rofi
+  else
+    echo "Neither dmenu(1) nor rofi(1) were found." >&2
+    exit 5
```
Comment (cdown) [inline clipmenu]: thanks for keeping distinct error codes :-)

File: clipmenu [hunk 4]
```diff
@@ -1,7 +1,17 @@
 #!/bin/bash
 
-: "${CM_LAUNCHER=dmenu}"
 : "${CM_DIR="${XDG_RUNTIME_DIR-"${TMPDIR-/tmp}"}"}"
+# Pick dmenu or rofi, depending on what's available on the system
```
Comment (cdown) [inline clipmenu]: this comment just states the obvious, you can remove it

**Comment by cdown** [comment]
Hi! This looks fine, only some minor things before merge.

File: clipmenu (hunk 3 above)
Comment (moviuro) [reply clipmenu]: Yeah, my bad, I wrote that in a hurry... I'll fix it tomorrow!

File: clipmenu (hunk 1 above)
Comment (moviuro) [reply clipmenu]: Why? This is further from POSIX syntax I'm used to. Is it faster?...

File: clipmenu (hunk 2 above)
Comment (moviuro) [reply clipmenu]: Yeah, my bad, I wrote that in a hurry... I'll fix it tomorrow!

File: clipmenu (hunk 1 above)
Comment (cdown) [reply clipmenu]: We already use lots of non-POSIX features, so trying to keep close compatibility isn't a goal. `-v` has clearer intent than `-z`. This isn't a massive deal but I'd rather stick to options which semantically represent the intent more clearly.

**Comment by cdown** [comment]
Hey! This still looks fine to me -- do you intend to update it for 6.0.0? :-)

**Comment by cdown** [comment]
Closing for now without prejudice -- happy to reopen if you intend to send an updated patch :-)

**Pull request closed without merging**